src/Simplex/Chat/Store/Messages.hs

```diff
@@ -1581,6 +1618,7 @@ getAllChatItems db vr user@User {userId} pagination search_ = do
       CPLast count -> liftIO $ getAllChatItemsLast_ count
       CPAfter afterId count -> liftIO . getAllChatItemsAfter_ afterId count . aChatItemTs =<< getAChatItem_ afterId
       CPBefore beforeId count -> liftIO . getAllChatItemsBefore_ beforeId count . aChatItemTs =<< getAChatItem_ beforeId
+      CPAround aroundId count -> liftIO . getAllChatItemsAround_ aroundId count . aChatItemTs =<< getAChatItem_ aroundId
```

Comment:
could also avoid re-reading middle item, by moving read into branches

Reply:
you may also need safeGetItem version though in map.. maybe not worth the refactoring, I think this api is not even used anywhere..